tests/unit/lib/samlib/test_resource_metadata_normalizer.py

```diff
@@ -456,6 +457,53 @@ def test_skip_normalizing_already_normalized_resource(self):
         self.assertEqual("new path", template_data["Resources"]["Function1"]["Properties"]["Code"])
         self.assertEqual("Function1", template_data["Resources"]["Function1"]["Metadata"]["SamResourceId"])
 
+    def test_referenced_data(self):
```

Comment:
Where are we asserting based on the new changes?

Reply:
Before this change, `SamResourceId` is assigned as `FirstFunction` for all resources due to the reason I explained in the PR desc.

After this change, all resources must have their logical id as `SamResourceId`, we are asserting them at the end of this changed block.

There is a redundant function that I forgot to remove, please ignore that, will be removing it now.